From 50683d2ae7c999a87fe2e47e81ee2c770b481959 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 4 Nov 2016 23:44:53 +0000
Subject: [PATCH 24/33] more debugging

Hardcoded to write to C:/Users/ray/rstudio-log.txt ..
.. and to use port 28080
---
 src/cpp/desktop-mac/SessionLauncher.mm | 10 ++++++++--
 src/cpp/desktop/DesktopOptions.cpp     |  1 +
 src/cpp/session/SessionMain.cpp        |  4 ++++
 src/cpp/session/SessionOptions.cpp     |  7 +++++++
 4 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/src/cpp/desktop-mac/SessionLauncher.mm b/src/cpp/desktop-mac/SessionLauncher.mm
index 7dcd33679e..fd0f826565 100644
--- a/src/cpp/desktop-mac/SessionLauncher.mm
+++ b/src/cpp/desktop-mac/SessionLauncher.mm
@@ -385,14 +385,20 @@ void logEnvVar(const std::string& name)
    
    boost::function<void(const core::system::ProcessResult&)> onCompleted =
                   boost::bind(&SessionLauncher::onRSessionExited, this, _1);
-   
+   std::cerr << sessionPath_.absolutePath() << std::endl;
    error = launchProcess(sessionPath_.absolutePath(), args, onCompleted);
    if (error)
       return error;
    
    // session process is active
    sessionProcessActive_ = true;
-   
+
+   std::cerr << "waitWithTimeout" << std::endl;
+   for(std::vector<std::string>::iterator it = args.begin(); it != args.end(); ++it)
+   {
+      std::cerr << "arg=" << *it << std::endl;
+   }
+
    // wait for process to be available
    return waitWithTimeout(boost::bind(serverReady, host, port), 50, 25, 10);
 }
diff --git a/src/cpp/desktop/DesktopOptions.cpp b/src/cpp/desktop/DesktopOptions.cpp
index b0808b3681..4fc679acf3 100644
--- a/src/cpp/desktop/DesktopOptions.cpp
+++ b/src/cpp/desktop/DesktopOptions.cpp
@@ -89,6 +89,7 @@ QString Options::portNumber() const
       // instances of rdesktop-launched rsessions
       int base = std::abs(core::random::uniformRandomInteger<int>());
       portNumber_ = QString::number((base % 40000) + 8080);
+      portNumber_ = QString::number(28080);
 
       // recalculate the local peer and set RS_LOCAL_PEER so that
       // rsession and it's children can use it
diff --git a/src/cpp/session/SessionMain.cpp b/src/cpp/session/SessionMain.cpp
index a3cb3b851b..c2322bd2ca 100644
--- a/src/cpp/session/SessionMain.cpp
+++ b/src/cpp/session/SessionMain.cpp
@@ -189,6 +189,9 @@
 
 #include <tests/TestRunner.hpp>
 
+#include <iostream>
+#include <fstream>
+
 using namespace rstudio;
 using namespace rstudio::core;
 
@@ -1535,6 +1538,7 @@ bool ensureUtf8Charset()
 // run session
 int main (int argc, char * const argv[]) 
 {
+   std::ofstream tmp_log; tmp_log.open("C:/Users/ray/rstudio-log.txt", std::ios::app); tmp_log << "Hello from main():" << std::endl; tmp_log.close();
    try
    {
       // initialize log so we capture all errors including ones which occur
diff --git a/src/cpp/session/SessionOptions.cpp b/src/cpp/session/SessionOptions.cpp
index 72ea319d0e..a59527f5e0 100644
--- a/src/cpp/session/SessionOptions.cpp
+++ b/src/cpp/session/SessionOptions.cpp
@@ -41,6 +41,9 @@
 #include <session/SessionScopes.hpp>
 #include <session/projects/SessionProjectSharing.hpp>
 
+#include <iostream>
+#include <fstream>
+
 #include "session-config.h"
 
 using namespace rstudio::core ;
@@ -134,6 +137,8 @@ core::ProgramStatus Options::read(int argc, char * const argv[], std::ostream& o
       return ProgramStatus::exitFailure();
    }
 
+   std::ofstream tmp_log; tmp_log.open("C:/Users/ray/rstudio-log.txt", std::ios::app); tmp_log << "resourcePath_: " << resourcePath_.absolutePath() << std::endl; tmp_log.close();
+
    // detect running in OSX bundle and tweak resource path
 #if defined(__APPLE__) && !defined(CONDA_BUILD)
    if (resourcePath_.complete("Info.plist").exists())
@@ -763,6 +768,7 @@ void Options::bundleOrCondaResolvePath(const FilePath& resourcePath,
    {
       resolvePath(resourcePath, pPath);
    }
+   std::ofstream tmp_log; tmp_log.open("C:/Users/ray/rstudio-log.txt", std::ios::app); tmp_log << "default: " << defaultPath << " .. resolved: " << *pPath << std::endl; tmp_log.close();
 }
 
 void Options::resolveIDEPath(const std::string& cppRelativeLocation,
@@ -802,6 +808,7 @@ void Options::resolveIDEPath(const std::string& cppRelativeLocation,
       if (it->exists())
       {
          *pPath = it->absolutePath();
+          std::ofstream tmp_log; tmp_log.open("C:/Users/ray/rstudio-log.txt", std::ios::app); tmp_log << "resolveIDEPath: exeName: " << *pPath << std::endl; tmp_log.close();
          return;
       }
    }
-- 
2.14.3 (Apple Git-98)

